--- configure	2006-12-02 18:27:35.000000000 -0500
+++ configure	2007-08-03 16:26:23.000000000 -0400
@@ -20236,52 +20236,52 @@
 # Setup for compiling build tools (fc-glyphname, etc)
 { echo "$as_me:$LINENO: checking for a C compiler for build tools" >&5
 echo $ECHO_N "checking for a C compiler for build tools... $ECHO_C" >&6; }
-if test $cross_compiling = yes; then
-  for ac_prog in gcc cc
-do
-  # Extract the first word of "$ac_prog", so it can be a program name with args.
-set dummy $ac_prog; ac_word=$2
-{ echo "$as_me:$LINENO: checking for $ac_word" >&5
-echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6; }
-if test "${ac_cv_prog_CC_FOR_BUILD+set}" = set; then
-  echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  if test -n "$CC_FOR_BUILD"; then
-  ac_cv_prog_CC_FOR_BUILD="$CC_FOR_BUILD" # Let the user override the test.
-else
-as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  test -z "$as_dir" && as_dir=.
-  for ac_exec_ext in '' $ac_executable_extensions; do
-  if { test -f "$as_dir/$ac_word$ac_exec_ext" && $as_executable_p "$as_dir/$ac_word$ac_exec_ext"; }; then
-    ac_cv_prog_CC_FOR_BUILD="$ac_prog"
-    echo "$as_me:$LINENO: found $as_dir/$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-done
-IFS=$as_save_IFS
-
-fi
-fi
-CC_FOR_BUILD=$ac_cv_prog_CC_FOR_BUILD
-if test -n "$CC_FOR_BUILD"; then
-  { echo "$as_me:$LINENO: result: $CC_FOR_BUILD" >&5
-echo "${ECHO_T}$CC_FOR_BUILD" >&6; }
-else
-  { echo "$as_me:$LINENO: result: no" >&5
-echo "${ECHO_T}no" >&6; }
-fi
-
-
-  test -n "$CC_FOR_BUILD" && break
-done
-
-else
+#if test $cross_compiling = yes; then
+#  for ac_prog in gcc cc
+#do
+#  # Extract the first word of "$ac_prog", so it can be a program name with args.
+#set dummy $ac_prog; ac_word=$2
+#{ echo "$as_me:$LINENO: checking for $ac_word" >&5
+#echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6; }
+#if test "${ac_cv_prog_CC_FOR_BUILD+set}" = set; then
+#  echo $ECHO_N "(cached) $ECHO_C" >&6
+#else
+#  if test -n "$CC_FOR_BUILD"; then
+#  ac_cv_prog_CC_FOR_BUILD="$CC_FOR_BUILD" # Let the user override the test.
+#else
+#as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
+#for as_dir in $PATH
+#do
+#  IFS=$as_save_IFS
+#  test -z "$as_dir" && as_dir=.
+#  for ac_exec_ext in '' $ac_executable_extensions; do
+#  if { test -f "$as_dir/$ac_word$ac_exec_ext" && $as_executable_p "$as_dir/$ac_word$ac_exec_ext"; }; then
+#    ac_cv_prog_CC_FOR_BUILD="$ac_prog"
+#    echo "$as_me:$LINENO: found $as_dir/$ac_word$ac_exec_ext" >&5
+#    break 2
+#  fi
+#done
+#done
+#IFS=$as_save_IFS
+#
+#fi
+#fi
+#CC_FOR_BUILD=$ac_cv_prog_CC_FOR_BUILD
+#if test -n "$CC_FOR_BUILD"; then
+#  { echo "$as_me:$LINENO: result: $CC_FOR_BUILD" >&5
+#echo "${ECHO_T}$CC_FOR_BUILD" >&6; }
+#else
+#  { echo "$as_me:$LINENO: result: no" >&5
+#echo "${ECHO_T}no" >&6; }
+#fi
+#
+#
+#  test -n "$CC_FOR_BUILD" && break
+#done
+#
+#else
   CC_FOR_BUILD=$CC
-fi
+#fi
 { echo "$as_me:$LINENO: result: $CC_FOR_BUILD" >&5
 echo "${ECHO_T}$CC_FOR_BUILD" >&6; }
 
--- configure.in	2007-08-03 16:21:30.000000000 -0400
+++ configure.in	2007-08-03 16:26:32.000000000 -0400
@@ -99,11 +99,11 @@
 
 # Setup for compiling build tools (fc-glyphname, etc)
 AC_MSG_CHECKING([for a C compiler for build tools])
-if test $cross_compiling = yes; then
-  AC_CHECK_PROGS(CC_FOR_BUILD, gcc cc)
-else
+#if test $cross_compiling = yes; then
+#  AC_CHECK_PROGS(CC_FOR_BUILD, gcc cc)
+#else
   CC_FOR_BUILD=$CC
-fi
+#fi
 AC_MSG_RESULT([$CC_FOR_BUILD])
 AC_SUBST(CC_FOR_BUILD)
 
