--- gettext-runtime/intl/localename.c	2006-11-27 12:02:00.000000000 -0500
+++ gettext-runtime/intl/localename.c	2007-07-04 09:40:08.000000000 -0400
@@ -1142,7 +1142,6 @@
 	switch (sub)
 	  {
 	  case SUBLANG_BENGALI_INDIA: return "bn_IN";
-	  case SUBLANG_BENGALI_BANGLADESH: return "bn_BD";
 	  }
 	return "bn";
       case LANG_BULGARIAN: return "bg_BG";
@@ -1356,7 +1355,6 @@
 	switch (sub)
 	  {
 	  case SUBLANG_PUNJABI_INDIA: return "pa_IN"; /* Gurmukhi script */
-	  case SUBLANG_PUNJABI_PAKISTAN: return "pa_PK"; /* Arabic script */
 	  }
 	return "pa";
       case LANG_RHAETO_ROMANCE: return "rm_CH";
@@ -1364,7 +1362,6 @@
 	switch (sub)
 	  {
 	  case SUBLANG_ROMANIAN_ROMANIA: return "ro_RO";
-	  case SUBLANG_ROMANIAN_MOLDOVA: return "ro_MD";
 	  }
 	return "ro";
       case LANG_RUSSIAN:
