diff -ur glib-2.26.0/glib/glib.symbols glib-2.26.0-new//glib/glib.symbols
--- glib-2.26.0/glib/glib.symbols	2010-09-27 09:17:56.000000000 -0400
+++ glib-2.26.0-new//glib/glib.symbols	2010-10-05 18:59:03.000000000 -0400
@@ -901,6 +901,7 @@
 g_option_context_add_main_entries
 g_option_error_quark
 g_option_context_free
+g_option_context_get_delocalize
 g_option_context_get_description
 g_option_context_get_help_enabled
 g_option_context_get_ignore_unknown_options
@@ -908,6 +909,7 @@
 g_option_context_get_summary
 g_option_context_new
 g_option_context_parse
+g_option_context_set_delocalize
 g_option_context_set_description
 g_option_context_set_help_enabled
 g_option_context_set_ignore_unknown_options
diff -ur glib-2.26.0/glib/goption.c glib-2.26.0-new//glib/goption.c
--- glib-2.26.0/glib/goption.c	2010-09-17 13:14:23.000000000 -0400
+++ glib-2.26.0-new//glib/goption.c	2010-10-05 19:04:41.000000000 -0400
@@ -197,6 +197,7 @@
 
   guint            help_enabled   : 1;
   guint            ignore_unknown : 1;
+  guint            delocalize : 1;
 
   GOptionGroup    *main_group;
 
@@ -336,6 +337,7 @@
   context->parameter_string = g_strdup (parameter_string);
   context->help_enabled = TRUE;
   context->ignore_unknown = FALSE;
+  context->delocalize = TRUE;
 
   return context;
 }
@@ -462,6 +464,23 @@
   return context->ignore_unknown;
 }
 
+void
+g_option_context_set_delocalize (GOptionContext *context,
+				 gboolean	 delocalize)
+{
+  g_return_if_fail (context != NULL);
+
+  context->delocalize = delocalize;
+}
+
+gboolean
+g_option_context_get_delocalize (GOptionContext *context)
+{
+  g_return_val_if_fail (context != NULL, FALSE);
+
+  return context->delocalize;
+}
+
 /**
  * g_option_context_add_group:
  * @context: a #GOptionContext
@@ -1151,7 +1170,10 @@
       {
         gchar *data;
 
-        data = g_locale_to_utf8 (value, -1, NULL, NULL, error);
+	if (context->delocalize)
+	    data = g_locale_to_utf8 (value, -1, NULL, NULL, error);
+	else
+	    data = g_strdup (value);
 
         if (!data)
           return FALSE;
@@ -1170,7 +1192,10 @@
       {
         gchar *data;
 
-        data = g_locale_to_utf8 (value, -1, NULL, NULL, error);
+	if (context->delocalize)
+	    data = g_locale_to_utf8 (value, -1, NULL, NULL, error);
+	else
+	    data = g_strdup (value);
 
         if (!data)
           return FALSE;
@@ -1203,7 +1228,10 @@
         gchar *data;
 
 #ifdef G_OS_WIN32
-        data = g_locale_to_utf8 (value, -1, NULL, NULL, error);
+	if (context->delocalize)
+	    data = g_locale_to_utf8 (value, -1, NULL, NULL, error);
+	else
+	    data = g_strdup (value);
 
         if (!data)
           return FALSE;
@@ -1226,7 +1254,10 @@
         gchar *data;
 
 #ifdef G_OS_WIN32
-        data = g_locale_to_utf8 (value, -1, NULL, NULL, error);
+	if (context->delocalize)
+	    data = g_locale_to_utf8 (value, -1, NULL, NULL, error);
+	else
+	    data = g_strdup (value);
 
         if (!data)
           return FALSE;
@@ -1283,7 +1314,10 @@
         else if (entry->flags & G_OPTION_FLAG_FILENAME)
           {
 #ifdef G_OS_WIN32
-            data = g_locale_to_utf8 (value, -1, NULL, NULL, error);
+	    if (context->delocalize)
+	      data = g_locale_to_utf8 (value, -1, NULL, NULL, error);
+	    else
+	      data = g_strdup (value);
 #else
             data = g_strdup (value);
 #endif
diff -ur glib-2.26.0/glib/goption.h glib-2.26.0-new//glib/goption.h
--- glib-2.26.0/glib/goption.h	2010-07-07 16:57:11.000000000 -0400
+++ glib-2.26.0-new//glib/goption.h	2010-10-05 18:59:03.000000000 -0400
@@ -321,6 +321,9 @@
 void		g_option_context_set_ignore_unknown_options (GOptionContext *context,
 							     gboolean	     ignore_unknown);
 gboolean        g_option_context_get_ignore_unknown_options (GOptionContext *context);
+void		g_option_context_set_delocalize (GOptionContext *context,
+							     gboolean	     ignore_unknown);
+gboolean        g_option_context_get_delocalize (GOptionContext *context);
 
 void            g_option_context_add_main_entries (GOptionContext      *context,
 						   const GOptionEntry  *entries,
