Index: autogen.sh
===================================================================
--- autogen.sh	(revision 2045)
+++ autogen.sh	(working copy)
@@ -61,11 +61,3 @@
 GNOME_DATADIR="$gnome_datadir"
 
 . $gnome_autogen
-
-if grep 'which gtkdoc-rebase >/dev/null &&' $srcdir/gtk-doc.make >/dev/null 2>&1; then
-    echo '----------------------------------------------------' 1>&2
-    echo "Your gtk-doc has a dependency problem.  Upgrade." 1>&2
-    echo "See http://bugzilla.gnome.org/show_bug.cgi?id=506506" 1>&2
-    echo '----------------------------------------------------' 1>&2
-    exit 1
-fi
