*** Makefile.am	2009-04-11 17:33:04.000000000 -0400
--- Makefile.am	2009-05-08 19:45:46.000000000 -0400
***************
*** 1,7 ****
  ## Makefile.am for GTK+
  include $(top_srcdir)/Makefile.decl
  
! SRC_SUBDIRS = util gdk gtk libgail-util modules demos tests testsuite examples
  SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
  
  # require automake 1.4
--- 1,7 ----
  ## Makefile.am for GTK+
  include $(top_srcdir)/Makefile.decl
  
! SRC_SUBDIRS = util gdk gtk libgail-util modules
  SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
  
  # require automake 1.4
