 gdk-pixbuf/Makefile.in	2010-03-22 23:32:58.000000000 -0400
--- modules/input/Makefile.in	2010-03-22 23:33:00.000000000 -0400
+++ modules/input/Makefile.in	2010-04-06 18:15:37.000000000 -0400
@@ -718,8 +718,8 @@
 @INCLUDE_IM_MULTIPRESS_FALSE@MULTIPRESS_MODULE = im-multipress.la
 imconffiledir = $(sysconfdir)/gtk-2.0
 dist_imconffile_DATA = im-multipress.conf
-@CROSS_COMPILING_FALSE@RUN_QUERY_IMMODULES_TEST = test -z "$(DESTDIR)"
-@CROSS_COMPILING_TRUE@RUN_QUERY_IMMODULES_TEST = false
+RUN_QUERY_IMMODULES_TEST = test -z "$(DESTDIR)"
+#RUN_QUERY_IMMODULES_TEST = false
 @BUILD_DYNAMIC_MODULES_TRUE@module_LTLIBRARIES = \
 @BUILD_DYNAMIC_MODULES_TRUE@	$(AM_ET_MODULE)			\
 @BUILD_DYNAMIC_MODULES_TRUE@	$(CEDILLA_MODULE)		\
@@ -1065,7 +1065,7 @@
 check-am: all-am
 	$(MAKE) $(AM_MAKEFLAGS) check-local
 check: check-am
-@CROSS_COMPILING_TRUE@all-local:
+#all-local:
 all-am: Makefile $(LTLIBRARIES) $(DATA) all-local
 installdirs:
 	for dir in "$(DESTDIR)$(moduledir)" "$(DESTDIR)$(imconffiledir)"; do \
