*** lib/Makefile.in	2009-03-25 22:14:26.000000000 -0400
--- lib/Makefile.in	2009-05-08 19:26:26.000000000 -0400
***************
*** 31,37 ****
  LIBTOOL_INSTALL = $(LIBTOOL) --mode=install
  LIBTOOL_UNINSTALL = $(LIBTOOL) --mode=uninstall
  # Windows resource compiler (windres). Used via libtool.
! RC = @RC@
  CP = cp
  MV = mv
  LN = @LN@
--- 31,37 ----
  LIBTOOL_INSTALL = $(LIBTOOL) --mode=install
  LIBTOOL_UNINSTALL = $(LIBTOOL) --mode=uninstall
  # Windows resource compiler (windres). Used via libtool.
! RC = @WINDRES@
  CP = cp
  MV = mv
  LN = @LN@
