diff -crN bzip2-1.0.5/Makefile bzip2-1.0.5-patched/Makefile
*** bzip2-1.0.5/Makefile	Thu Feb 14 07:39:18 2008
--- bzip2-1.0.5-patched/Makefile	Sun Nov 23 17:57:14 2008
***************
*** 15,31 ****
  SHELL=/bin/sh
  
  # To assist in cross-compiling
! CC=gcc
! AR=ar
! RANLIB=ranlib
  LDFLAGS=
  
  BIGFILES=-D_FILE_OFFSET_BITS=64
  CFLAGS=-Wall -Winline -O2 -g $(BIGFILES)
  
  # Where you want it installed when you do 'make install'
! PREFIX=/usr/local
  
  
  OBJS= blocksort.o  \
        huffman.o    \
--- 15,33 ----
  SHELL=/bin/sh
  
  # To assist in cross-compiling
! #CC=gcc
! #AR=ar
! #RANLIB=ranlib
  LDFLAGS=
  
  BIGFILES=-D_FILE_OFFSET_BITS=64
  CFLAGS=-Wall -Winline -O2 -g $(BIGFILES)
  
  # Where you want it installed when you do 'make install'
! PREFIX=$(DESTDIR)/$(prefix)
  
+ RC =$(WINDRES)
+ RCFLAGS = --define GCC_WINDRES
  
  OBJS= blocksort.o  \
        huffman.o    \
***************
*** 33,56 ****
        randtable.o  \
        compress.o   \
        decompress.o \
        bzlib.o
  
! all: libbz2.a bzip2 bzip2recover test
  
! bzip2: libbz2.a bzip2.o
! 	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2 bzip2.o -L. -lbz2
  
  bzip2recover: bzip2recover.o
  	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2recover bzip2recover.o
  
! libbz2.a: $(OBJS)
! 	rm -f libbz2.a
! 	$(AR) cq libbz2.a $(OBJS)
! 	@if ( test -f $(RANLIB) -o -f /usr/bin/ranlib -o \
! 		-f /bin/ranlib -o -f /usr/ccs/bin/ranlib ) ; then \
! 		echo $(RANLIB) libbz2.a ; \
! 		$(RANLIB) libbz2.a ; \
! 	fi
  
  check: test
  test: bzip2
--- 35,58 ----
        randtable.o  \
        compress.o   \
        decompress.o \
+       bz2librc.o   \
        bzlib.o
  
! SHAREDLIB = libbz2.dll
! IMPLIB    = libbz2.dll.a
  
! all: bzip2 bzip2recover test $(SHAREDLIB) $(IMPLIB)
! 
! bzip2: libbz2.dll bzip2.o
! 	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2 bzip2.o -L. $(IMPLIB)
  
  bzip2recover: bzip2recover.o
  	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2recover bzip2recover.o
  
! $(IMPLIB): $(SHAREDLIB)
! 
! $(SHAREDLIB) $(IMPLIB): $(OBJS)
! 	$(CC) -shared -Wl,--out-implib -Wl,$(IMPLIB) -o $@ $(OBJS)
  
  check: test
  test: bzip2
***************
*** 72,115 ****
  install: bzip2 bzip2recover
  	if ( test ! -d $(PREFIX)/bin ) ; then mkdir -p $(PREFIX)/bin ; fi
  	if ( test ! -d $(PREFIX)/lib ) ; then mkdir -p $(PREFIX)/lib ; fi
- 	if ( test ! -d $(PREFIX)/man ) ; then mkdir -p $(PREFIX)/man ; fi
- 	if ( test ! -d $(PREFIX)/man/man1 ) ; then mkdir -p $(PREFIX)/man/man1 ; fi
  	if ( test ! -d $(PREFIX)/include ) ; then mkdir -p $(PREFIX)/include ; fi
- 	cp -f bzip2 $(PREFIX)/bin/bzip2
- 	cp -f bzip2 $(PREFIX)/bin/bunzip2
- 	cp -f bzip2 $(PREFIX)/bin/bzcat
- 	cp -f bzip2recover $(PREFIX)/bin/bzip2recover
- 	chmod a+x $(PREFIX)/bin/bzip2
- 	chmod a+x $(PREFIX)/bin/bunzip2
- 	chmod a+x $(PREFIX)/bin/bzcat
- 	chmod a+x $(PREFIX)/bin/bzip2recover
- 	cp -f bzip2.1 $(PREFIX)/man/man1
- 	chmod a+r $(PREFIX)/man/man1/bzip2.1
  	cp -f bzlib.h $(PREFIX)/include
  	chmod a+r $(PREFIX)/include/bzlib.h
! 	cp -f libbz2.a $(PREFIX)/lib
! 	chmod a+r $(PREFIX)/lib/libbz2.a
! 	cp -f bzgrep $(PREFIX)/bin/bzgrep
! 	ln -s -f $(PREFIX)/bin/bzgrep $(PREFIX)/bin/bzegrep
! 	ln -s -f $(PREFIX)/bin/bzgrep $(PREFIX)/bin/bzfgrep
! 	chmod a+x $(PREFIX)/bin/bzgrep
! 	cp -f bzmore $(PREFIX)/bin/bzmore
! 	ln -s -f $(PREFIX)/bin/bzmore $(PREFIX)/bin/bzless
! 	chmod a+x $(PREFIX)/bin/bzmore
! 	cp -f bzdiff $(PREFIX)/bin/bzdiff
! 	ln -s -f $(PREFIX)/bin/bzdiff $(PREFIX)/bin/bzcmp
! 	chmod a+x $(PREFIX)/bin/bzdiff
! 	cp -f bzgrep.1 bzmore.1 bzdiff.1 $(PREFIX)/man/man1
! 	chmod a+r $(PREFIX)/man/man1/bzgrep.1
! 	chmod a+r $(PREFIX)/man/man1/bzmore.1
! 	chmod a+r $(PREFIX)/man/man1/bzdiff.1
! 	echo ".so man1/bzgrep.1" > $(PREFIX)/man/man1/bzegrep.1
! 	echo ".so man1/bzgrep.1" > $(PREFIX)/man/man1/bzfgrep.1
! 	echo ".so man1/bzmore.1" > $(PREFIX)/man/man1/bzless.1
! 	echo ".so man1/bzdiff.1" > $(PREFIX)/man/man1/bzcmp.1
  
  clean: 
! 	rm -f *.o libbz2.a bzip2 bzip2recover \
  	sample1.rb2 sample2.rb2 sample3.rb2 \
  	sample1.tst sample2.tst sample3.tst
  
--- 74,89 ----
  install: bzip2 bzip2recover
  	if ( test ! -d $(PREFIX)/bin ) ; then mkdir -p $(PREFIX)/bin ; fi
  	if ( test ! -d $(PREFIX)/lib ) ; then mkdir -p $(PREFIX)/lib ; fi
  	if ( test ! -d $(PREFIX)/include ) ; then mkdir -p $(PREFIX)/include ; fi
  	cp -f bzlib.h $(PREFIX)/include
  	chmod a+r $(PREFIX)/include/bzlib.h
! 	cp -f $(SHAREDLIB) $(PREFIX)/bin
! 	chmod a+r $(PREFIX)/bin/$(SHAREDLIB)
! 	cp -f $(IMPLIB) $(PREFIX)/lib
! 	chmod a+r $(PREFIX)/lib/$(IMPLIB)
  
  clean: 
! 	rm -f *.o libbz2.dll* bzip2 bzip2recover \
  	sample1.rb2 sample2.rb2 sample3.rb2 \
  	sample1.tst sample2.tst sample3.tst
  
***************
*** 132,138 ****
  	$(CC) $(CFLAGS) -c bzip2.c
  bzip2recover.o: bzip2recover.c
  	$(CC) $(CFLAGS) -c bzip2recover.c
! 
  
  distclean: clean
  	rm -f manual.ps manual.html manual.pdf
--- 106,113 ----
  	$(CC) $(CFLAGS) -c bzip2.c
  bzip2recover.o: bzip2recover.c
  	$(CC) $(CFLAGS) -c bzip2recover.c
! bz2librc.o: libbz2.rc
! 	$(RC) $(RCFLAGS) -o $@ $<
  
  distclean: clean
  	rm -f manual.ps manual.html manual.pdf
diff -crN bzip2-1.0.5/bzip2.c bzip2-1.0.5-patched/bzip2.c
*** bzip2-1.0.5/bzip2.c	Sun Dec  9 07:22:06 2007
--- bzip2-1.0.5-patched/bzip2.c	Sun Nov 23 17:57:14 2008
***************
*** 128,134 ****
  #if BZ_LCCWIN32
  #   include <io.h>
  #   include <fcntl.h>
! #   include <sys\stat.h>
  
  #   define NORETURN       /**/
  #   define PATH_SEP       '\\'
--- 128,134 ----
  #if BZ_LCCWIN32
  #   include <io.h>
  #   include <fcntl.h>
! #   include <sys/stat.h>
  
  #   define NORETURN       /**/
  #   define PATH_SEP       '\\'
diff -crN bzip2-1.0.5/libbz2.rc bzip2-1.0.5-patched/libbz2.rc
*** bzip2-1.0.5/libbz2.rc	Wed Dec 31 19:00:00 1969
--- bzip2-1.0.5-patched/libbz2.rc	Sun Nov 23 18:18:09 2008
***************
*** 0 ****
--- 1,39 ----
+ #include <windows.h>
+ 
+ #ifdef GCC_WINDRES
+ VS_VERSION_INFO		VERSIONINFO
+ #else
+ VS_VERSION_INFO		VERSIONINFO	MOVEABLE IMPURE LOADONCALL DISCARDABLE
+ #endif
+   FILEVERSION		1,0,5,0
+   PRODUCTVERSION	1,0,5,0
+   FILEFLAGSMASK		VS_FFI_FILEFLAGSMASK
+ #ifdef _DEBUG
+   FILEFLAGS		1
+ #else
+   FILEFLAGS		0
+ #endif
+   FILEOS		VOS_DOS_WINDOWS32
+   FILETYPE		VFT_DLL
+   FILESUBTYPE		0	// not used
+ BEGIN
+   BLOCK "StringFileInfo"
+   BEGIN
+     BLOCK "040904E4"
+     //language ID = U.S. English, char set = Windows, Multilingual
+     BEGIN
+       VALUE "FileDescription",	"bzip2 data compression library\0"
+       VALUE "FileVersion",	"1.0.5\0"
+       VALUE "InternalName",	"libbz2.dll\0"
+       VALUE "LegalCopyright",	"(C) 1996-2007 Julian Seward <jseward@bzip.org>\0"
+       VALUE "OriginalFilename",	"libbz2.dll\0"
+       VALUE "ProductName",	"libbz2\0"
+       VALUE "ProductVersion",	"1.0.5\0"
+       VALUE "Comments","DLL support by Alessandro Iacopetti & Gilles Vollant\0"
+     END
+   END
+   BLOCK "VarFileInfo"
+   BEGIN
+     VALUE "Translation", 0x0409, 1252
+   END
+ END
