diff -cr configure.in configure.in
*** configure.in	2007-10-14 08:15:21.000000000 -0400
--- configure.in	2008-07-07 13:07:14.000000000 -0400
***************
*** 119,142 ****
  ALL_LINGUAS="ar az be ca cs da de dz el en_CA en_GB es eu fa fi fr ga gl hr hu it ja ka ko lt ml ms nb ne nl oc pl pt pt_BR ru rw sk sl sq sr sr@Latn sv ta tr uk vi zh_CN zh_HK zh_TW"
  AM_GLIB_GNU_GETTEXT
  
  dnl *******************************
  dnl GConf schemas
  dnl ******************************
! AC_PATH_PROG(GCONFTOOL, gconftool-2)
! AM_GCONF_SOURCE_2
! 
! dnl ******************************
! dnl Checks for GConf librarie
! dnl ******************************
! 
! GCONF_MODULES="gconf-2.0"
! PKG_CHECK_MODULES(GCONF, $GCONF_MODULES, have_gconf=yes, have_gconf=no)
! AM_CONDITIONAL(HAVE_GCONF, test x"$have_gconf" = "xyes")
! if test x"$have_gconf" = "xyes"
! then
! 	GCONF_CFLAGS="$GCONF_CFLAGS -DHAVE_GCONF"
!         LIBGNOMEDB_REQUIRES="$LIBGNOMEDB_REQUIRES $GCONF_MODULES"
! fi
  AC_SUBST(GCONF_CFLAGS)
  AC_SUBST(GCONF_LIBS)
  
--- 119,139 ----
  ALL_LINGUAS="ar az be ca cs da de dz el en_CA en_GB es eu fa fi fr ga gl hr hu it ja ka ko lt ml ms nb ne nl oc pl pt pt_BR ru rw sk sl sq sr sr@Latn sv ta tr uk vi zh_CN zh_HK zh_TW"
  AM_GLIB_GNU_GETTEXT
  
+ dnl The following conditional is set in AM_GCONF_SOURCE_2.
+ dnl Because we may skip its execution, we have to set a default here.
+ AM_CONDITIONAL([GCONF_SCHEMAS_INSTALL], [false])
+ m4_pattern_allow([^AM_GCONF_SOURCE_2$])
+ 
  dnl *******************************
  dnl GConf schemas
  dnl ******************************
! 
! GCONFTOOL=
! HAVE_GCONF=no
! AM_CONDITIONAL(WITH_GCONF, false)
! AM_CONDITIONAL(HAVE_GCONF, false)
! GCONF_CFLAGS=
! GCONF_LIBS=
  AC_SUBST(GCONF_CFLAGS)
  AC_SUBST(GCONF_LIBS)
  
diff -cr data/Makefile.am data/Makefile.am
*** data/Makefile.am	2007-09-01 08:28:25.000000000 -0400
--- data/Makefile.am	2008-07-07 13:01:35.000000000 -0400
***************
*** 15,22 ****
  SUBDIRS = $(glade3_directory_16) $(glade3_directory_22)
  
  # GConf schemas
  schemadir   = $(GCONF_SCHEMA_FILE_DIR)
! schema_DATA = libgnomedb-3.0.schemas
  
  @INTLTOOL_SCHEMAS_RULE@
  
--- 15,25 ----
  SUBDIRS = $(glade3_directory_16) $(glade3_directory_22)
  
  # GConf schemas
+ schemas = libgnomedb-3.0.schemas
  schemadir   = $(GCONF_SCHEMA_FILE_DIR)
! if GCONF_SCHEMAS_INSTALL
! schema_DATA = $(schemas)
! endif
  
  @INTLTOOL_SCHEMAS_RULE@
  
