--- autogen.sh	2003-11-11 15:41:37.000000000 -0500
+++ autogen.sh	2009-10-10 07:14:00.000000000 -0400
@@ -39,7 +39,4 @@
 echo "Creating configure..."
 autoconf$AUTOCONF_VERSION
 
-echo ""
-echo "You can run ./configure now."
-echo ""
-
+./configure $*
--- lib/psiconv/configuration.c	2005-11-15 10:08:08.000000000 -0500
+++ lib/psiconv/configuration.c	2007-08-06 16:45:43.000000000 -0400
@@ -336,7 +336,7 @@
   pathptr = path;
   while (strlen(pathptr)) {
     /* Isolate the next filename */
-    filename_len = (index(pathptr,':')?(index(pathptr,':') - pathptr):
+    filename_len = (strchr(pathptr,':')?(strchr(pathptr,':') - pathptr):
 	                               strlen(pathptr));
     filename = malloc(filename_len + 1);
     filename = strncpy(filename,pathptr,filename_len);
--- formats/generate_xhtml.sh	2004-02-23 08:18:57.000000000 -0500
+++ formats/generate_xhtml.sh	2009-10-10 06:49:15.000000000 -0400
@@ -26,7 +26,7 @@
 echo "Generating xhtml docs..."
 
 libtool=$basedir/libtool
-psiconv=$basedir/program/psiconv/psiconv
+psiconv=$basedir/program/psiconv/psiconv.exe
 indexfile=$outputdir/index
 tempdir=$outputdir/.temp
 mkindex=$basedir/formats/index_html.sh
